tests/android/tests/test_android_smoke_profile_edit_functionality.py

```diff
@@ -0,0 +1,169 @@
+            expect(edit_profile_page.text_select_from_gallery).to_have("Select from gallery")
+            expect(edit_profile_page.change_profile_image_cancel_button).to_exist()
+            expect(edit_profile_page.change_profile_image_cancel_button_text).to_have("Cancel")
+            edit_profile_page.change_profile_image_cancel_button.click()
+
+        try:
+            with allure.step("click on Done button"):
+                edit_profile_page.done_button.click()
+                edit_profile_page.done_button.wait_to_disappear(timeout=5)
+
+            with allure.step("click on back button"):
+                edit_profile_page.back_navigation_button.click()
+                expect(profile_page.profile_settings_about_me_body).to_have("Automation Test")
+
+            with allure.step("click on edit profile button"):
+                profile_page.edit_profile_button.click()
+                expect(edit_profile_page.profile_tf_select_location).to_have(values.CANADA)
+                expect(edit_profile_page.select_spoken_language).to_have(values.FRENCH)
+                expect(edit_profile_page.about_me_input).to_have("Automation Test")
+
+        except Exception as exception:
+            test_exception = exception
+            setup_logging.info(f"Following exception occurred: {exception}")
+            setup_logging.info("Performing cleanup")
+        finally:
+            with allure.step("Change values to original"):
+                if profile_page.edit_profile_button.exists(raise_exception=False):
+                    profile_page.edit_profile_button.click()
+                edit_profile_page.profile_tf_select_location.click()
+                edit_profile_page.sb_search_field.send_keys(values.PROFILE_LOCATION_US)
+                country_opt_canada = Globals.get_country_by_resource_id(CountryAbbreviation.US)
+                expect(country_opt_canada).to_have(values.PROFILE_LOCATION_US)
+                country_opt_canada.click()
+                expect(edit_profile_page.profile_tf_select_location).to_have(values.PROFILE_LOCATION_US)
+                edit_profile_page.select_spoken_language.click()
+                edit_profile_page.sb_search_field.send_keys(values.ENGLISH)
+                language_opt_english = Globals.get_language_by_abbreviation(LanguageAbbreviation.EN)
+                expect(language_opt_english).to_have(values.ENGLISH)
+                language_opt_english.click()
+                expect(edit_profile_page.select_spoken_language).to_have(values.ENGLISH)
+
+                edit_profile_page.about_me_input.click()
+                edit_profile_page.about_me_input.send_keys(values.PROFILE_ABOUT_ME_TEXT)
+                edit_profile_page.done_button.click()
+                edit_profile_page.android_loading_circle.wait_to_disappear(20)
+
+            with allure.step("click on back button"):
+                edit_profile_page.back_navigation_button.click()
```

Comment:
As I mentioned earlier, this is the ideal use case for the post-test-case dedicated method.